### DISCUSSION TITLE
Casting a ray against specific bodies / ragdoll

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#556] [Q&A] [answered]
Is there a way to do a raycast (or I guess any query) against a specific set of bodies? Basically skipping the entire broadphase.

In my case I want to apply an impulse to a ragdoll but I would first need to figure out on which body part of the ragdoll to apply the impulse and at which location. Since I am only interested in the ragdoll, I thought I could skip doing a full world raycast and directly operate on just those bodies.

**Comment by jrouwe** [comment, marked as answer]
I would do something like:

```
ClosestHitCollisionCollector<CastRayCollector> collector;
for (Body &body : bodies_of_ragdoll)
   body.GetTransformedShape().CastRay(..., collector);
```

**Reply by jankrassnigg** [reply]
Ah that's how to do it, thanks!
